pkg/asset/installconfig/installconfig.go

```diff
@@ -142,3 +155,70 @@ func (a *InstallConfig) setDefaults() error {
 	defaults.SetInstallConfigDefaults(a.Config)
 	return nil
 }
+
+func loadDeprecated(data []byte, config *types.InstallConfig) error {
```

Comment:
This is fairly compact, and we want to support it at some point, but it's not a strict requirement at the moment, with @crawford saying:

> We don't have to maintain backward compat for a beta release. We can just go to v1beta2 and tell people to update.

So if it helps review, we may want to spin this off into its own PR.

Reply:
This is just in there temporarily so that this PR can land without breaking hive and byo in the near term. If it is spun off into its own PR, then that other PR would need to land before this PR.